src/Parsers/IdentifierQuotingStyle.h

```diff
@@ -14,4 +14,10 @@ enum class IdentifierQuotingStyle : uint8_t
     BackticksMySQL,  /// `mysql` style, most same as Backticks, but it uses '``' to escape '`'
 };
 
+enum class IdentifierQuotingRule : uint8_t
+{
+    UserDisplay,    /// When the identifiers is one of the certain keywords defined in `writeProbablyQuotedStringImpl`
+    WhenNecessary,  /// When the identifiers is one of the certain keywords defined in `writeProbablyQuotedStringImpl`, and ambiguous identifiers passed to `writeIdentifier`
+    Always,         /// Always quote identifiers
```

Comment:
We should add these descriptions to the [documentation](https://clickhouse.com/docs/en/operations/settings/settings) as well (see settings.md)